wagtail/admin/views/reports/workflows.py

```diff
@@ -198,19 +215,16 @@ def get_queryset(self):
     def decorate_paginated_queryset(self, object_list):
         return [obj for obj in object_list if obj.content_object]
 
-    def dispatch(self, request, *args, **kwargs):
-        if not page_permission_policy.user_has_any_permission(
-            request.user, ["add", "change", "publish"]
-        ):
-            raise PermissionDenied
-        return super().dispatch(request, *args, **kwargs)
-
 
 class WorkflowTasksView(ReportView):
-    template_name = "wagtailadmin/reports/workflow_tasks.html"
-    title = _("Workflow tasks")
+    results_template_name = "wagtailadmin/reports/workflow_tasks_results.html"
+    page_title = _("Workflow tasks")
     header_icon = "thumbtack"
     filterset_class = WorkflowTasksReportFilterSet
+    index_url_name = "wagtailadmin_reports:workflow_tasks"
+    index_results_url_name = "wagtailadmin_reports:workflow_tasks_results"
+    permission_policy = page_permission_policy
+    any_permission_required = ["add", "change", "publish"]
```

Comment:
Notice how this view previously didn't have a `dispatch()` override to check the permissions. The fix in 13048eab793d80f6bc7a32aa14a1c3511f494fa9 was incomplete as it only added the `dispatch()` override in `WorkflowView` but not `WorkflowTasksView`. The menu item uses the same `WorkflowReportMenuItem` class, so it will be hidden as expected. However, the view is still accessible via direct URL (but no data will be shown, so it's safe-ish), as reported in the original issue https://github.com/wagtail/wagtail/issues/8845#issuecomment-1184781500